data/helpers.d/mysql

```diff
@@ -90,3 +90,47 @@ ynh_mysql_create_user() {
 ynh_mysql_drop_user() {
     ynh_mysql_execute_as_root "DROP USER '${1}'@'localhost';"
 }
+
+# Create a database, an user and its password. Then store the password in the app's config
+#
+# User of database will be stored in db_user's variable.
+# Name of database will be stored in db_name's variable.
+# And password in db_pwd's variable.
+#
+# usage: ynh_mysql_generate_db user name
+# | arg: user - Owner of the database
+# | arg: name - Name of the database
+ynh_mysql_generate_db () {
+	db_pwd=$(ynh_string_random)	# Generate a random password
+	ynh_mysql_create_db "$2" "$1" "$db_pwd"	# Create the database
+	ynh_app_setting_set $app mysqlpwd $db_pwd	# Store the password in the app's config
+}
+
+# Remove a database if it exists, and the associated user
+#
+# usage: ynh_mysql_remove_db user name
+# | arg: user - Owner of the database
+# | arg: name - Name of the database
+ynh_mysql_remove_db () {
+	if mysqlshow -u root -p$(sudo cat $MYSQL_ROOT_PWD_FILE) | grep -q "^| $2"; then	# Check if the database exists
+		echo "Remove database $2" >&2
+		ynh_mysql_drop_db $2	# Remove the database
+		ynh_mysql_drop_user $1	# Remove the associated user to database
+	else
+		echo "Database $2 not found" >&2
+	fi
+}
+
+# Correct the name given in argument for mariadb
+#
+# Avoid invalid name for your database
+#
+# Exemple: dbname=$(ynh_make_valid_dbid $app)
+#
+# usage: ynh_make_valid_dbid name
+# | arg: name - name to correct
+# | ret: the corrected name
+ynh_make_valid_dbid () {
```

Comment:
Imho "generate" feels more like creating something 'out of nothing'. Here we just want to "correct"/"sanitize"/"fix" the input